File: Contest/A_Twice.py
import sys
input = sys.stdin.readline

t = int(input())

for _ in range(t):
    n = int(input())
    arr = list(map(int, input().split()))
    
    freq = [0] * (n+1)
    
    for num in arr:
        freq[num] += 1
        
    score = 0
    
    for i in range(1, n+1):
        score += freq[i] // 2
        
    print(score)

File: Contest/A_Twice.kt
fun main() {
    val t = readLine()!!.toInt()
    
    repeat(t) {
        val n = readLine()!!.toInt()
        val arr = readLine()!!.split(" ").map { it.toInt() }
        
        val freq = IntArray(n + 1)
        
        for (num in arr) {
            freq[num]++
        }
        
        var score = 0
        
        for (f in freq) {
            score += f / 2
        }
        
        println(score)
    }
}